alpha/apps/kaltura/lib/multiLingualUtils.php

```diff
@@ -245,12 +246,31 @@ protected static function getMappingForDefaultLanguage(&$responseObject, &$multi
 		}
 	}
 	
+	/*
+	 * Return an array containing all the languages that are mapped in the entry, even if a language is mapped to part of the fields.
+	 * If a language appears in the mapping, the entry supports it
+	 */
+	protected static function getMappedLanguages($newMultiLingualMapping, $supportedFields)
+	{
+		$mappedLanguages = array();
+		foreach ($supportedFields as $field)
+		{
+			array_push($mappedLanguages, array_keys($newMultiLingualMapping[$field]));
+		}
+		
+		return array_unique($mappedLanguages)[0];
+	}
+	
 	protected static function setRequestedLanguageStringInField(&$responseObject, $dbObject, $newMultiLingualMapping, $requestLanguage = null, KalturaDetachedResponseProfile $responseProfile = null)
 	{
 		$defaultLanguage = self::getDefaultLanguage($dbObject);
 		$language = ($requestLanguage && $defaultLanguage) ? $requestLanguage : $defaultLanguage;
 		$supportedFields = $dbObject->getMultiLingualSupportedFields();
 		$supportedFieldsInRequestedLang = array();
+		$mappedLanguages = self::getMappedLanguages($newMultiLingualMapping, $supportedFields);
+		
+		$isLanguageMapped = !(empty($mappedLanguages)) && in_array($language, $mappedLanguages);
```

Comment:
prefer to do it with if statement and initialize  with the false value before isLanguageMapped